btree/btree.c

```diff
@@ -4619,21 +4630,33 @@ static int64_t btree_get_kv_tick(struct m0_sm_op *smop)
 		 *  If valid sibling found, return first key of the sibling
 		 *  subtree else return key not exist.
 		 */
-		if (bop->bo_flags & BOF_EQUAL) {
-			if (oi->i_key_found)
-				node_rec(&s);
-			else
-				s.s_rec.r_flags = M0_BSC_KEY_NOT_FOUND;
-		} else {
-			if (lev->l_idx < node_count(lev->l_node))
-				node_rec(&s);
-			else {
-				rc = btree_sibling_first_key_get(oi, tree, &s);
-				if (rc != 0) {
-					node_op_fini(&oi->i_nop);
-					return fail(bop, rc);
+		if (bop->bo_opc == M0_BO_GET) {
+			if (bop->bo_flags & BOF_EQUAL) {
+				if (oi->i_key_found)
+					node_rec(&s);
+				else
+					s.s_rec.r_flags = M0_BSC_KEY_NOT_FOUND;
+			} else {
+				if (lev->l_idx < count)
+					node_rec(&s);
+				else {
+					rc = btree_sibling_first_key(oi, tree,
+								     &s);
+					if (rc != 0) {
+						node_op_fini(&oi->i_nop);
+						return fail(bop, rc);
+					}
 				}
 			}
+		} else {
+			/** MIN/MAX key operation. */
+			if (count > 0) {
+				s.s_idx = bop->bo_opc == M0_BO_MINKEY ? 0 :
+					  count - 1;
+				node_rec(&s);
+			} else
+				/** Only root node is present and is empty. */
+				s.s_rec.r_flags = M0_BSC_KEY_NOT_FOUND;
 		}
 
 		bop->bo_cb.c_act(&bop->bo_cb, &s.s_rec);
```

Comment:
Looks like you will call the callback even if there is nothing in the tree !!! We need to change this later to avoid calling the callback in such error scenarios.